synapse/config/oidc_config.py

```diff
@@ -568,3 +598,6 @@ class OidcProviderConfig:
 
     # the config of the user mapping provider
     user_mapping_provider_config = attr.ib()
+
+    # required attributes to require in userinfo to allow login/registration
+    attribute_requirements = attr.ib(type=Iterable[SsoAttributeRequirement])
```

Comment:
I think this can be defined as a list here and that's fine (a list is an iterable):

```suggestion
    attribute_requirements = attr.ib(type=List[SsoAttributeRequirement])
```